### ISSUE TITLE
Offer a way for Query.bind() to ignore inline parameters

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15575]
The `Query.bind(int, Object)` method exposes a weird historic behaviour, where we don't distinguish between inline parameters and actual bind values. This can be seen with the following test code:

```java
println(ctx.select(inline(1).eq(1)).getSQL());
println(ctx.select(inline(1).eq(1)).bind(1, 2).getSQL());
```

It prints:

```sql
select (1 = ?)
select (2 = ?)
```

So, the 1-based bind index `1` allows for overriding the `inline(1)` value, not the `eq(1)` value (which is convenience for `eq(val(1))`. This is quite unexpected for some users, especially when they parse their SQL:

```java
println(DSL.using(dialect)
    .parser()
    .parseResultQuery("SELECT COUNT(1) FROM T WHERE C = ?")
    .bind(2, "x") // C = ? has index 2, not 1!
    .getSQL(ParamType.INLINED)
);
```

In other words, it works quite differently from plain JDBC. The problem is deeper than it just appears. If we have to emulate things, and we duplicate a bind value, then the bind index doesn't correspond to the *input* index, but to the output index.

The problem is that index based `bind()` calls use the `Query.getParams()` method, which is a name based approach, so there's a mismatch here.

For the folks who use the parser, it's probably better to supply the bind values at parse time:

```java
println(DSL.using(dialect)
    .parser()
    .parseResultQuery("SELECT COUNT(1) FROM T WHERE C = ?", "x")
    .getSQL(ParamType.INLINED)
);
```

**Comment by lukaseder** [comment]
The complications around this method, along with the fact that users have to be aware of *every* jOOQ emulation (e.g. `LOG(base, number)` turns into `LOG(number, base)` for some dialects, effectively reversing bind indexes. There are also duplications, or inlinings of bind values for some emulations) makes me think if we shouldn't just rather deprecate this method and stop offering it.

The valid use-cases for binding variables outside of the query are:

- Plain SQL templates (though there, it's better to place the bind values directly with the template)
- Parser (though there agani, it's better to place the bind values directly with the parse call)

I'll think about this a bit more.

**Comment by lukaseder** [comment]
Related: https://github.com/jOOQ/jOOQ/issues/11513
